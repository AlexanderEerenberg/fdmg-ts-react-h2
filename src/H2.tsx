import * as React from 'react';

/**
 * H2 compononent renders a heading
 */
export default class H2 extends React.Component {
    public state: any;
    public props: any;

    constructor(props: any) {
        super(props);
        this.props = props;
        console.info('H2 component', props);
    }

    componentDidMount() {
        console.info('H2 mounted');
    }
        
    componentWillUnmount() {
        console.info('H2 will unmount');
    }

    componentWillReceiveProps(nextProps) {
        console.info('H2 next props', nextProps);
    }

    render() {
        return (
            <h2 className={this.props.className}>{this.props.text}</h2>
        );
    }
}
