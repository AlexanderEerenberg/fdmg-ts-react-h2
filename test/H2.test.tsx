import * as React from 'react';
import * as renderer from 'react-test-renderer';
import H2 from '../src/H2';

console.info = function() {};

test('H2 renders correctly', () => {
    const component = renderer.create(
        <H2 text={'test h2'}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

test('H2 renders correctly with CSS class', () => {
    const component = renderer.create(
        <H2 text={'test h2'} className={'title'}/>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});