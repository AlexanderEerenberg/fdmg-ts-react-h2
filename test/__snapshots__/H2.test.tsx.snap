// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 renders correctly 1`] = `
<h2
  className={undefined}
>
  test h2
</h2>
`;

exports[`H2 renders correctly with CSS class 1`] = `
<h2
  className="title"
>
  test h2
</h2>
`;
